extensions/windows-event-log/CollectorInitiatedSubscription.cpp

```diff
@@ -0,0 +1,789 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "CollectorInitiatedSubscription.h"
+#include <vector>
+#include <queue>
+#include <map>
+#include <vector>
+#include <set>
+#include <sstream>
+#include <stdio.h>
+#include <string>
+#include <iostream>
+#include <memory>
+#include <codecvt>
+
+#include "io/DataStream.h"
+#include "core/ProcessContext.h"
+#include "core/ProcessSession.h"
+#include "utils/ScopeGuard.h"
+
+#pragma comment(lib, "wevtapi.lib")
+#pragma comment(lib, "Wecapi.lib")
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace processors {
+
+#define LOG_SUBSCRIPTION_ERROR(error) logError(__LINE__, error)
+#define LOG_SUBSCRIPTION_WINDOWS_ERROR(info) logWindowsError(__LINE__, info)
+
+struct ISupportedProperty {
+  void virtual Init(const std::shared_ptr<core::ProcessContext>& context) = 0;
+};
+
+class SuportedProperties {
+  std::vector<ISupportedProperty*> vISupportedProperty_;
+  std::set<core::Property> sSupportedProperties_;
+
+public:
+  template <typename T>
+  void add(T& t)
+  {
+    vISupportedProperty_.emplace_back(&t);
+    sSupportedProperties_.insert(t);
+  }
+
+  void init(const std::shared_ptr<core::ProcessContext>& context) {
+    for (auto pProp : vISupportedProperty_) {
+      pProp->Init(context);
+    }
+  }
+
+  operator std::set<core::Property>() const
+  {
+    return sSupportedProperties_;
+  }
+};
+
+static SuportedProperties s_supportedProperties;
+
+template <typename T>
+class SupportedProperty: public ISupportedProperty, public core::Property {
+  T t_;
+
+public:
+  template <typename ...Args>
+  SupportedProperty(const Args& ...args) : core::Property(args...) {
+    s_supportedProperties.add(*this);
+  }
+
+  void Init(const std::shared_ptr<core::ProcessContext>& context) override;
+
+  T value() {
+    return t_;
+  }
+};
+
+void SupportedProperty<std::wstring>::Init(const std::shared_ptr<core::ProcessContext>& context) {
+  std::string val;
+  context->getProperty(getName(), val);
+
+  t_ = std::wstring(val.begin(), val.end());
+};
+
+void SupportedProperty<uint64_t>::Init(const std::shared_ptr<core::ProcessContext>& context) {
+  context->getProperty(getName(), t_);
+};
+
+struct TimeProperty {
+  int64_t time_{};
+
+  operator int64_t() {
+    return time_;
+  }
+};
+
+void SupportedProperty<TimeProperty>::Init(const std::shared_ptr<core::ProcessContext>& context) {
+  std::string strTime;
+  context->getProperty(getName(), strTime);
+
+  int64_t time{};
+  core::TimeUnit unit;
+  core::Property::StringToTime(strTime, time, unit);
+  core::Property::ConvertTimeUnitToMS(time, unit, time);
```

Comment:
For those that are TimePeriodValue you can do context->getProperty(MinAge.getName(), uint64_t); and the conversion will be performed for you